lib/tasks/development_replication.rake

```diff
@@ -0,0 +1,14 @@
+namespace :development do
+  namespace :replication do
+    desc "Manually setup logical replication locally"
+    task :setup => :environment do
+      TaskHelpers::Development::Replication.backup
+      TaskHelpers::Development::Replication.setup
+      TaskHelpers::Development::Replication.restore
```

Comment:
To ensure that restore happens on failure and to mirror what teardown is doing, it might be cleaner for this to just call setup, and then up above something like:

```ruby
        def backup
          if File.exist?(guid_file)
            FileUtils.rm_f(backup_guid)
            FileUtils.mv(guid_file, backup_guid)
          end
        end

        def restore
          if File.exist?(backup_guid)
            FileUtils.rm_f(guid_file)
            FileUtils.mv(backup_guid, guid_file)
          end
        end

        def setup
          backup

          REMOTES.each {|r| setup_remote(r) }
          setup_global(GLOBAL)

          # TODO: We have the technology to watch for this and report when it's all good or bad
          puts "Local replication is setup... try checking for users in all regions: psql -U root development_replication_99 -c \"select id from users;\""
        ensure
          restore
        end

        def teardown
          REMOTES.each do |r|
            teardown_global_subscription_for_region(r)
            teardown_remote_publication(r)
          end
        end
```

Reply:
yeah, that's much nicer